hooks/post_gen_project.py

```diff
@@ -1,9 +1,137 @@
+
+            another_q = "Would you like to add another team (y/n)? [n]: "
+            if not check_choice(another_q):
+                break
+
+    return collaborators
+
+def run_terraform(directory):
+    terraform = Terraform(directory)
+    terraform.init()
+    ret_code, _, _ = terraform.apply(auto_approve=True, capture_output=False)
+
+    return ret_code
+
+def remove_terraform_files(directory):
+    file_names = ["main.tf", "variables.tf", "terraform.tfvars", "terraform.tfstate"]
+    for file_name in file_names:
+        os.remove(directory +"/"+ file_name)
+    shutil.rmtree(directory)
 
 if __name__ == "__main__":
-{% if cookiecutter.create_repo == "yes" %}
-    subprocess.check_call(['git', 'init'])
-    subprocess.check_call(['hub', 'create'])
-{%- else %}
-    pass
-{% endif %}
+    if "{{ cookiecutter.create_repo }}".lower() == "yes":
+        subprocess.check_call(['git', 'init'])
+        subprocess.check_call(['hub', 'create'])
+    elif "{{ cookiecutter.create_org_repo }}".lower() == "yes":
+        # Yes/No Questions
+        YN_QUESTIONS = {
+            "enable_stict_checks": "Require branches to be up to date before merging (y/n)? [n]: ",
+            "enforce_admins": "Require admins to oblige by status checks (y/n)? [n]: ",
+            "enforce_code_owner_review": "Require a designated code owner to approve pull requests (y/n)? [n]: ",
+            "enable_issues": "Enable the GitHub Issues features on the repository (y/n)? [n]: ",
+            "enable_merge_commits": "Allow merge commits (y/n)? [n]: ",
+            "enable_squash_merging": "Allow squash merging (y/n)? [n]: ",
+            "enable_rebase_merging": "Allow rebase merging (y/n)? [n]: ",
+        }
+
+        # Multiple entry questions
+        DICT_QUESTIONS = {
+            "topics": {
+                "start_question": "Add a topic to the repository (y/n)? [n]: ",
+                "init_loop": "What topic would you like to add? "
+            },
+            "required_status_checks": {
+                "start_question": "Add required checks for merging into the branch (y/n)? [n]: ",
+                "init_loop": "What check would you like to add? "
+            },
+            "restricted_pr_teams": {
+                "start_question": "Restrict which teams can dismiss push requests (y/n)? [n]: ",
+                "init_loop": "What team would you like to restrict? "
+            },
+            "restricted_push_teams": {
+                "start_question": "Restrict who can push to the branch (y/n)? [n]: ",
+                "init_loop": "What team would you like to restrict? "
+            }
+        }
+
+        directory = "terraform/"
+        token_q = "Enter the personal access token for creating the organizational repo: "
+        os.environ['GITHUB_ORGANIZATION'] = "{{ cookiecutter.github_username }}"
+        os.environ['GITHUB_TOKEN'] = raw_input(token_q).lower()
+
+        with open( directory  + "/terraform.tfvars", "w") as f:
```

Comment:
terraform supports json tfvars, no need to do this string parsing. just dump the vars as json.

* https://www.terraform.io/docs/configuration/variables.html#variable-definitions-tfvars-files

Reply:
I've attempted to just dump the GITHUB_ORGANIZATION and GITHUB_TOKEN into the terraform.tfvars file but often had issues with it and only thing that would consistently work is when I had it in the environment.

Reply:
eh, more referring to everything _below_ this `with` statement. hadn't even looked at the GITHUB envs... i think that's just an artifact of how github is displaying the snippet around the comment on the line...